examples/ChannelsController.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Channel;
+use App\User;
+use Cmgmyr\Messenger\Models\Message;
+use Cmgmyr\Messenger\Models\Thread;
+use Illuminate\Support\Facades\Auth;
+
+class ChannelsController extends MessagesController
+{
+    protected $viewSpace = 'channels';
+    protected $routeSpace = 'channels';
+
+    /**
+     * Show all of the message threads to the user.
+     *
+     * @return mixed
+     */
+    public function index()
+    {
+        // All threads, ignore deleted/archived participants
+//         $threads = Thread::getAllLatest()->get();
+
+        // All threads that user is participating in
+//         $threads = Thread::forUser(Auth::id())->latest('updated_at')->get();
+
+        // All threads that user is participating in, with new messages
+        // $threads = Thread::forUserWithNewMessages(Auth::id())->latest('updated_at')->get();
+
+        $threads = Thread::publicThread(Channel::first())->latest('updated_at')->get();
```

Comment:
Where is the `publicThread` scope method? And cause of the polymorph relation the query should look like this:

```php
$threads = Channel::first()->threads()->byIsPublic()->latest('updated_at')->get();
```

```php
public function scopeByIsPublic(Builder $query, $isPublic = true)
{
    $query->where('is_public', $isPublic);
}
```

Reply:
somehow missed that in my compare.. put in there now.. I named the method a more clear name and added the query builder you sampled